crates/oxc_parser/src/lib.rs

```diff
@@ -71,7 +71,7 @@ impl<'a> Parser<'a> {
             errors,
             token: Token::default(),
             prev_token_end: 0,
-            state: ParserState::default(),
+            state: ParserState::new(oxc_allocator::Vec::new_in(allocator)),
```

Comment:
@Boshen is this OK? We can't pass the `AstBuilder` instance into `ParserState::new` because of the lifetimes, this was the workaround I ended up with 👍🏼